token_dispenser/token_dispenser_lambda.py

```diff
@@ -0,0 +1,137 @@
+from cryptography.hazmat.primitives.serialization.pkcs12 import load_key_and_certificates
+from cryptography.hazmat.backends import default_backend
+
+from token_dispenser.aws.s3 import download_s3_file
+from token_dispenser.aws.secret_manager import get_secret_value
+from token_dispenser.aws.launchpad_token import get_token
+import token_dispenser.configuration as config
+from token_dispenser.repository.token_repo import put_token, get_token_by_client_id
+import json, re
+import logging
+from token_dispenser.logging_config import initialize_logger, shared_logger
+import time
+
+def decode_pkcs12(p12_file_path, password: str):
+    """
+    Decodes a PKCS#12 file to extract the private key, certificate, and additional certificates.
+
+    :param p12_file_path: Path to the PKCS#12 file (.p12 or .pfx).
+    :param password: Password for the PKCS#12 file (if any).
+    :return: A tuple containing the private key, certificate, and additional certificates.
+    """
+    with open(p12_file_path, "rb") as file:
+        pkcs12_data = file.read()
+
+    # Decode the PKCS#12 file
+    private_key, certificate, additional_certs = load_key_and_certificates(
+        pkcs12_data,
+        password.encode("utf-8") if password else None,
+        backend=default_backend()
+    )
+    return private_key, certificate, additional_certs
+
+
+def get_new_token(client_id:str):
+    """
+    obtain a new token based on client_id and write the newly obtained token into DynamoDB
+
+    :param client_id: a required field which is used as key to cache the token in dynamoDB.
+    :return: a json object containing the token data.
+    """
+    try:
+        p12_file = download_s3_file(bucket_name=config.LAUNCHPAD_PFX_FILE_S3_BUCKET,
+                                            key=config.LAUNCHPAD_PFX_FILE_S3_KEY, local_storage_dir='/tmp')
+        shared_logger().info(f"p12 file downloaded from s3 successfully to: {p12_file}")
+        password = get_secret_value(config.LAUNCHPAD_PFX_PASSWORD_SECRET_ARN)
+        private_key, cert, additional_certs = decode_pkcs12(p12_file, password)
+        shared_logger().info(f"cert files decoded successfully")
+        # Create launchpad token
+        token_json = get_token(url=config.LAUNCHPAD_GETTOKEN_URL, private_key=private_key,
+                                                        certificate=cert)
+        put_token(client_id, json.dumps(token_json), int(token_json['expires_at']))
+        return token_json
+    except Exception as e:
+        print(f"Failed on get_new_token process: {e}")
+        raise e
+
+
+def satisfy_minimum_alive_secs(token_json:json, minimum_alive_secs:int) -> bool:
```

Comment:
Why do we need to pass in the entire token here instead of just the requested minimum alive secs?

Also, the correct type would be `dict`